lib/ansible/modules/commands/xmlrpc_client.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Stefan Midjich <swehack at gmail dot com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'curated'
+}
+
+DOCUMENTATION = '''
+---
+module: xmlrpc_client
+
+short_description: Run XMLRPC commands
+
+version_added: "2.4"
+
+description:
+    - This module lets you establish an XMLRPC session and call any method
+      by its xmlrpc path. On dry run this module will only establish API
+      connection and check if your method exists in the method list but not
+      call it.
+
+options:
+    url:
+        description:
+            - The API URL you connect to.
+        required: true
+    path:
+        description:
+            - The RPC path/method you are calling.
+        required: true
+    args:
+        description:
+            - A list of arguments for the RPC call. Placed first in any RPC call.
+        required: false
+    kwargs:
+        description:
+            - A dictionary of arguments for the RPC call. Comes after the args list.
+        required: false
+
+author:
+    - Stefan Midjich (@stemid)
+'''
+
+EXAMPLES = '''
+---
+
+vars:
+  api_url: https://api.localhost:9002
+
+tasks:
+  # Establish session
+  - name: login to api
+    xmlrpc_client:
+      url: "{{api_url}}"
+      path: login
+      args:
+        - admin
+        - secret password.
+    register: sid
+
+  # Use session to execute privileged command. In this case server.group.add
+  # returns a group ID from the API.
+  - name: create new server group
+    xmlrpc_client:
+      url: "{{api_url}}"
+      path: server.group.add
+      args:
+        - "{{sid.returned}}"
+        - Dev environment web server group - Sweden, Malmoe
+        - web
+    register: server_group_id
+    when: sid|changed
+
+  # Use previous value in more API calls.
+  - name: create new server
+    xmlrpc_client:
+      url: "{{api_url}}"
+      path: server.add
+      args:
+        - "{{sid.returned}}"
+        - vm-web01
+        - 80
+        - "{{server_group_id.returned}}"
+    register: server_id
+    when: server_group_id.returned
+'''
+
+RETURN = '''
+returned:
+  description: The output value that the RPC call returns
+  type: string
+  returned: success, changed
+'''
+
+
+try:
+    from xmlrpc.client import ServerProxy
+except ImportError:
+    from xmlrpclib import ServerProxy
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    module_args = dict(
+        url=dict(type='str', required=True),
+        path=dict(type='str', required=True),
+        args=dict(type='list', required=False, default=[]),
+        kwargs=dict(type='dict', required=False, default={})
```

Comment:
Here also `required=False` is not needed for parameters that are not required. Guideline dictates to leave it out.